znvis/visualizer/visualizer.py

```diff
@@ -553,6 +625,92 @@ def _update_particles(self, visualizer=None, step: int = None):
 
         visualizer.post_redraw()  # re-draw the window.
 
+    def _update_particles_back(self, visualizer=None, step: int = None):
+        """
+        Update the positions of the particles one step back (rewind-feature)
+
+        Parameters
+        ----------
+        step : int
+                Step to update to.
+
+        Returns
+        -------
+        Updates the positions of the particles in the box.
+        """
+        if visualizer is None:
+            visualizer = self.vis
+        if step is None:
+            if self.counter == 0:
+                self.counter = self.number_of_steps - 1
+            else:
+                self.counter -= 1
+            step = self.counter
+        self._draw_particles(visualizer=visualizer)  # draw the particles.
+
+        # draw the vector field if it exists.
+        if self.vector_field is not None:
+            self._draw_vector_field(visualizer=visualizer) 
+
+        visualizer.post_redraw()  # re-draw the window.
+
+
```

Comment:
There are too many spaces between these methods. This is something that will be picked up in the linting. However, we do have pre-commit hooks which you can install with `precommit install .` from the ZnVis directory. This will prevent you from committing failing code.